kpatch-build/kpatch-elf.c

```diff
@@ -88,17 +88,29 @@ struct section *find_section_by_index(struct list_head *list, unsigned int index
 	return NULL;
 }
 
-struct section *find_section_by_name(struct list_head *list, const char *name)
+struct section *find_nth_section_by_name( struct list_head *list, int nth, const char *name)
 {
 	struct section *sec;
 
-	list_for_each_entry(sec, list, list)
-		if (!strcmp(sec->name, name))
-			return sec;
+	if (!list || !list->next || !name)
```

Comment:
(I know nearly nothing about kpatch. Drive-by comment...)

The condition looks strange. I believe it's not this function job to check `!list || !name`. It should be a precondition calling this function.

`!list->next` can be subsumed by the loop below.